### DISCUSSION TITLE
failed to describe index

### DISCUSSION THREAD

**Discussion by huangxueliang** [milvus-io/milvus#37535] [Q&A and General discussion] [answered]
[2024/11/08 06:11:10.158 +00:00] [WARN] [datacoord/metrics_info.go:69] [\"failed to describe index, ignore to report index metrics\"] [traceID=1c18b1c77ee979c446fa7242bf07e5c8] [collection=451875630419352409] [error=\"index not found[indexName=]\"]\n","stream":"stdout","time":"2024-11-08T06:11:10.158377303Z"}
{"log":"[2024/11/08 06:11:13.159 +00:00] [WARN] [datacoord/index_service.go:696] [\"DescribeIndex fail\"] [traceID=5eaf3535094151c2e73473a0448b5f32] [collectionID=451875630419352409] [indexName=] [error=\"index not found[indexName=]\"]\n","stream":"stdout","time":"2024-11-08T06:11:13.159365132Z"}
{"log":"[2024/11/08 06:11:13.159 +00:00] [WARN] [datacoord/metrics_info.go:69] [\"failed to describe index, ignore to report index metrics\"] [traceID=5eaf3535094151c2e73473a0448b5f32] [collection=451875630419352409] [error=\"index not found[indexName=]\"]\n","stream":"stdout","time":"2024-11-08T06:11:13.159661134Z"}
{"log":"[2024/11/08 06:11:16.158 +00:00] [WARN] [datacoord/index_service.go:696] [\"DescribeIndex fail\"] [traceID=8a01c35c63a83d8cb2f74f0390ddab7f] [collectionID=451875630419352409] [indexName=] [error=\"index not found[indexName=]\"]\n","stream":"stdout","time":"2024-11-08T06:11:16.158451272Z"}
{"log":"[2024/11/08 06:11:16.158 +00:00] [WARN] [datacoord/metrics_info.go:69] [\"failed to describe index, ignore to report index metrics\"] [traceID=8a01c35c63a83d8cb2f74f0390ddab7f] [collection=451875630419352409] [error=\"index not found[indexName=]\"]\n","stream":"stdout","time":"2024-11-08T06:11:16.15881603Z"}

how can slove the issue?

**Comment by yhmo** [comment]
To describe an index, provide an index name:

```
from pymilvus import MilvusClient, DataType

collection_name = "XXX"
dim = 128

client = MilvusClient(uri='http://localhost:19530')

schema = client.create_schema(
    auto_id=False,
    enable_dynamic_schema=True,
)

schema.add_field(field_name="id", datatype=DataType.INT64, is_primary=True)
schema.add_field(field_name="vector", datatype=DataType.FLOAT_VECTOR, dim=dim)

index_params = client.prepare_index_params()
index_params.add_index(field_name="vector", index_type="FLAT", metric_type="COSINE", params={})

client.drop_collection(collection_name=collection_name)
client.create_collection(collection_name=collection_name, schema=schema, index_params=index_params)
print(client.describe_index(collection_name=collection_name, index_name="vector"))
```

By default, if you didn't specify an index name, the field name will be applied as index name.

The output:
`{'index_type': 'FLAT', 'metric_type': 'COSINE', 'field_name': 'vector', 'index_name': 'vector', 'total_rows': 0, 'indexed_rows': 0, 'pending_index_rows': 0, 'state': 'Finished'}`


If you specify a name for the index:
```
index_params = client.prepare_index_params()
index_params.add_index(field_name="vector", index_type="FLAT", metric_type="COSINE", params={}, index_name="aaaaa")

client.drop_collection(collection_name=collection_name)
client.create_collection(collection_name=collection_name, schema=schema, index_params=index_params)
print(client.describe_index(collection_name=collection_name, index_name="aaaaa"))
```

The output:
`{'index_type': 'FLAT', 'metric_type': 'COSINE', 'field_name': 'vector', 'index_name': 'aaaaa', 'total_rows': 0, 'indexed_rows': 0, 'pending_index_rows': 0, 'state': 'Finished'}`

**Reply by huangxueliang** [reply]
is does means all the collections need to create index?

**Reply by yhmo** [reply, marked as answer]
Each vector field must have an index: https://milvus.io/docs/index-vector-fields.md?tab=floating#Index-Vector-Fields

Try the first example to know the basic workflow: https://github.com/milvus-io/pymilvus/blob/master/examples/hello_milvus.py
